hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterClientProtocol.java

```diff
@@ -545,12 +567,12 @@ public boolean complete(String src, String clientName, ExtendedBlock last,
             long.class},
         new RemoteParam(), clientName, last, fileId);
 
+    final List<RemoteLocation> locations = rpcServer.getLocationsForPath(src, true);
     if (last != null) {
-      return rpcClient.invokeSingle(last, method, Boolean.class);
+      RemoteLocation location = getLocationWithBPID(locations, last.getBlockPoolId());
+      return rpcClient.invokeSingle(location, method, Boolean.class);
```

Comment:
We are doing this `` rpcServer.getLocationsForPath(src, true);`` and then getting the location for I think every old invokeSingle with Block as an argument.
Can we not do this logic inside the removed ``invokeSingle`` method?